hostplumber/pkg/utils/sriov/sriov.go

```diff
@@ -0,0 +1,309 @@
+	err := filepath.Walk(devicePath, func(path string, info os.FileInfo, err error) error {
+		name := info.Name()
+
+		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			isVF, _ := filepath.Match("virtfn*", name)
+			if isVF == false {
+				return nil
+			}
+			fullPath, _ := filepath.EvalSymlinks(path)
+			if err := SetDriverForVf(fullPath, driver); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		// TODO: Why am I not returning an error here?
+		return
+	}
+}
+
+func GetPfListForVendorAndDevice(vendor, device string) ([]string, error) {
+	// This function will return any IF that matches vendor and device ID
+	// If optional PCI address is given, it must match
+	var matchingPfs []string
+	err := filepath.Walk(consts.SysClassNet, func(path string, info os.FileInfo, err error) error {
+		ifName := info.Name()
+		// Everything in this directory is a symlink
+		ifPath, _ := filepath.EvalSymlinks(path)
+		if ifPath == consts.SysClassNet {
+			// Skip the root /sys/class/net folder
+			return nil
+		}
+		if ifName == "lo" {
+			return nil
+		}
+
+		devicePathLink := filepath.Join(ifPath, "device")
+		devicePath, _ := filepath.EvalSymlinks(devicePathLink)
+		deviceIdFile := filepath.Join(devicePath, "device")
+
+		if _, err := os.Stat(deviceIdFile); err != nil {
+			return nil
+		}
+		fd, err := os.Open(deviceIdFile)
+		if err != nil {
+			return err
+		}
+		var deviceId string
+		_, err = fmt.Fscanf(fd, "0x%s\n", &deviceId)
+		if err != nil {
+			return err
+		}
+
+		vendorIdFile := filepath.Join(devicePath, "vendor")
+		if _, err := os.Stat(vendorIdFile); err != nil {
+			return nil
+		}
+		fd, err = os.Open(vendorIdFile)
+		if err != nil {
+			return err
+		}
+		var vendorId string
+		_, err = fmt.Fscanf(fd, "0x%s\n", &vendorId)
+		if err != nil {
+			return err
+		}
+
+		if vendor == vendorId && device == deviceId {
+			matchingPfs = append(matchingPfs, ifName)
+		}
+		return nil
+	})
+
+	if len(matchingPfs) == 0 {
+		err = errors.NewBadRequest("Failed to find any devices matching vendor and device ID")
+	}
+	return matchingPfs, err
+}
+
+func GetPfNameForPciAddr(pciAddr string) (string, error) {
+	var matchingPf string
+	err := filepath.Walk(consts.SysClassNet, func(path string, info os.FileInfo, err error) error {
+		ifName := info.Name()
+		// Everything in this directory is a symlink
+		ifPath, _ := filepath.EvalSymlinks(path)
+		if ifPath == consts.SysClassNet {
+			return nil
+		}
+		if ifName == "lo" {
+			return nil
+		}
+
+		devicePathLink := filepath.Join(ifPath, "device")
+		devicePath, _ := filepath.EvalSymlinks(devicePathLink)
+
+		devPci := filepath.Base(devicePath)
+		if devPci == pciAddr {
+			matchingPf = ifName
+			// Short circuit because we found a match
+			return io.EOF
+		}
+		return nil
+	})
+	if err != nil {
+		if err == io.EOF {
+			err = nil
+		}
+	} else {
+		err = errors.NewBadRequest("Failed to find interface with matching PCI address")
+	}
```

Comment:
I am not completely sure what this function is meant to do, but it seems to me that the error handling is not correct here. 

Shouldn't it be?:

```suggestion
	if err != nil {
		if err == io.EOF {
			err = nil
		} else {
		        err = errors.NewBadRequest("Failed to find interface with matching PCI address")
	        }
	} 
```